js/app/test/blocks_chained_events.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { test, expect } from "@gradio/tootils";
+
+test("Success event", async ({ page }) => {
+	const textbox = page.getByLabel("Result");
+	await expect(textbox).toHaveValue("");
+
+	await Promise.all([
+		page.click("text=Trigger Failure"),
+		page.waitForResponse("**/run/predict")
+	]);
+	// Since the event is not triggered, the value should not change
+	expect(textbox).toHaveValue("");
+
+	await Promise.all([
+		page.click("text=Trigger Success"),
+		page.waitForResponse("**/run/predict")
+	]);
+
+	expect(textbox).toHaveValue("Success event triggered");
+});
+
+test("Consecutive success event", async ({ page }) => {
+	const textbox = page.getByLabel("Consecutive Event");
+
+	await page.click("text=Trigger Consecutive Success");
+	// need to wait for all the events to fire
+	await page.waitForTimeout(1000);
```

Comment:
Not sure myself, but is there a better way than to have timeouts? I wonder if we might end up with flaky tests as a result of using timeouts